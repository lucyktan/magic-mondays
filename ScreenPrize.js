import React, { Component } from 'react';
import {
  StyleSheet,
  View,
  Text,
  TouchableHighlight,
  Image
} from 'react-native';

class ScreenPrize extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    const navigation = this.props.navigation;
    return (
      <View style={styles.container}>
          <Image
            source={require('./img/congratulations.gif')}
            style={{flex: 1, width: 300, height: 100, position: 'absolute', top: 20}}
          />

          <Image
            source={require('./img/prize_beautybox.png')}
            style={{flex: 2, width: 300, height: 340, position: 'absolute'}}
          />

          <Text style={{fontSize: 13, color: 'gray', position: 'absolute', marginLeft: 20, marginRight: 20, bottom: 90}}>A chic pouch of 5 deluxe beauty samples + 1 bonus and $5 off your next beauty purchase</Text>


        <View style={styles.buttonContainer}>
          <TouchableHighlight
            onPress={() => navigation.navigate("ScreenWallet", { screen: "Screen Wallet" })}
            style={[styles.button, {backgroundColor: '#E22130'}]}>
            <Text style={styles.buttonText}>Save Prize to Wallet</Text>
          </TouchableHighlight>
        </View>
      </View>
    );
  }
};

const styles = StyleSheet.create({
  container: {
    flex: 2,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#ffffff',
    height: 300
  },
  titleText: {
    fontSize: 22
  },
  buttonContainer: {
    flexDirection: 'row',
    marginLeft: 20,
    marginRight: 20,
    position: 'absolute',
    bottom: 20
  },
  button: {
    height: 50,
    flex: 2,
    justifyContent: 'center'
  },
  buttonText: {
    color: 'white',
    alignSelf: 'center',
    fontSize: 18
  }
});
export default ScreenPrize;