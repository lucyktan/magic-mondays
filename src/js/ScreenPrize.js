import React, { Component } from 'react';
import {
  StyleSheet,
  View,
  Text,
  TouchableHighlight,
  Image,
  Dimensions
} from 'react-native';
import congratulationImage from '../img/congratulations.gif';
import prizeImage from '../img/prize_beautybox.png';
import floatingStars from '../assets/floatingstars.mov';
import { Video } from 'expo';

class ScreenPrize extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    const navigation = this.props.navigation;
  
    return (
      <View style={styles.container}>
          <Video
            source={require('../assets/floatingstars.mov')}
            rate={1.0}
            volume={0.0}
            muted={true}
            resizeMode={"cover"}
            shouldPlay
            isLooping
            style={styles.video}
          />

          <Image
            source={congratulationImage}
            style={styles.congratsImage}
          />

          <Image resizeMode = 'cover'
            source={prizeImage}
            style={styles.prizeImage}
          />

          <Text style={styles.prizeDesc}>A chic pouch of 5 deluxe beauty samples + 1 bonus and $5 off your next beauty purchase</Text>

        <View style={styles.buttonContainer}>
          <TouchableHighlight
            onPress={() => navigation.navigate("ScreenWallet", { screen: "Screen Wallet" })}
            style={[styles.button, {backgroundColor: '#E22130'}]}>
            <Text style={styles.buttonText}>Save Prize to Wallet</Text>
          </TouchableHighlight>
        </View>
      </View>
    );
  }
};

const styles = StyleSheet.create({
  container: {
    flex: 2,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#ffffff',
    height: 300
  },
  video: {
    position: 'absolute',
    top: 0,
    left: 0,
    bottom: 0,
    right: 0,
  },
  congratsImage: {
    flex: 1, 
    width: 600, 
    height: 100, 
    position: 'absolute', 
    top: 20
  },
  prizeImage: {
    flex: 2, 
    width: 450,
    height: 300,
    position: 'absolute',
    margin: 'auto',
    overflow: 'visible'
  },
  prizeDesc: {
    fontSize: 13, 
    color: 'gray', 
    alignSelf: 'center',
    position: 'absolute', 
    marginLeft: 20, 
    marginRight: 20, 
    bottom: 150
  },
  titleText: {
    fontSize: 22
  },
  buttonContainer: {
    flexDirection: 'row',
    marginLeft: 20,
    marginRight: 20,
    position: 'absolute',
    bottom: 20
  },
  button: {
    height: 50,
    flex: 2,
    justifyContent: 'center'
  },
  buttonText: {
    color: 'white',
    alignSelf: 'center',
    fontSize: 18
  }
});
export default ScreenPrize;