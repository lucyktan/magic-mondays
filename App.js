import React from 'react';
// import { StyleSheet, Text, View } from 'react-native';
import {StackNavigator,} from 'react-navigation';

import ScreenHome from './src/js/ScreenHome';
import ScreenSplash from './src/js/ScreenSplash';
import ScreenGame from './src/js/ScreenGame';
import ScreenPrize from './src/js/ScreenPrize';
import ScreenWallet from './src/js/ScreenWallet';


const App = StackNavigator({
  ScreenHome: {screen: ScreenHome},
  ScreenSplash: {screen: ScreenSplash},
  ScreenGame: {screen: ScreenGame},
  ScreenPrize: {screen: ScreenPrize},
  ScreenWallet: {screen: ScreenWallet},
},
{
  headerMode: 'none',
  navigationOptions: {
    headerVisible: false,
  }
});
export default App;
