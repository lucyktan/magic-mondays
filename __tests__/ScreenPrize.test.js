import 'react-native';
import React from 'react';
import {Video} from 'expo';
import ScreenPrize from '../src/js/ScreenPrize';
import App from '../App';

import {configure, shallow, mount } from 'enzyme';

import renderer from 'react-test-renderer';

it('renders correctly', () => {

    let navigation = {
    state: {
        key: 'id-1512512594799-2',
        params: {
	      screen: 'Screen Game',
        user: 'prize'
  	  },
        routeName: 'ScreenGame',
      }
    };
  const tree = renderer.create(<ScreenPrize navigation={navigation} />).toJSON();
  expect(tree).toMatchSnapshot();
});
