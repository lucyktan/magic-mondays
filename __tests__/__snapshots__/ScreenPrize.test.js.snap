// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ffffff",
      "flex": 2,
      "height": 300,
      "justifyContent": "center",
    }
  }
>
  <Image
    defaultSource={undefined}
    onError={undefined}
    onLoad={undefined}
    onLoadEnd={undefined}
    onLoadStart={undefined}
    onProgress={undefined}
    resizeMode="cover"
    source={1}
    style={
      Array [
        Object {
          "height": NaN,
          "width": NaN,
        },
        Object {
          "flex": 1,
          "height": 100,
          "position": "absolute",
          "top": 20,
          "width": 300,
        },
      ]
    }
  />
  <Image
    defaultSource={undefined}
    onError={undefined}
    onLoad={undefined}
    onLoadEnd={undefined}
    onLoadStart={undefined}
    onProgress={undefined}
    resizeMode="cover"
    source={1}
    style={
      Array [
        Object {
          "height": NaN,
          "width": NaN,
        },
        Object {
          "flex": 2,
          "height": 340,
          "position": "absolute",
          "width": 300,
        },
      ]
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "bottom": 90,
        "color": "gray",
        "fontSize": 13,
        "marginLeft": 20,
        "marginRight": 20,
        "position": "absolute",
      }
    }
  >
    A chic pouch of 5 deluxe beauty samples + 1 bonus and $5 off your next beauty purchase
  </Text>
  <View
    style={
      Object {
        "bottom": 20,
        "flexDirection": "row",
        "marginLeft": 20,
        "marginRight": 20,
        "position": "absolute",
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Array [
            Object {
              "flex": 2,
              "height": 50,
              "justifyContent": "center",
            },
            Object {
              "backgroundColor": "#E22130",
            },
          ],
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "alignSelf": "center",
            "color": "white",
            "fontSize": 18,
          }
        }
      >
        Save Prize to Wallet
      </Text>
    </View>
  </View>
</View>
`;
